src/app/jobs/job.component.html

```diff
@@ -0,0 +1,98 @@
+<!--
+Copyright 2018 Novo Nordisk Foundation Center for Biosustainability, DTU.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+<div class="container" *ngIf="job">
+  <h1>
+    <a routerLink="/jobs">Jobs</a> / #{{ job.id }}
+  </h1>
+
+  <mat-list>
+    <mat-list-item>
```

Comment:
I don't agree, this "duplication" is all encapsulated in a single place and doesn't come with the risks or downsides that duplicated code normally entails.

It may become relevant to display values differently, say f.ex. by adding a link for the chosen product, which would belong in the template, not the datastructure. Pipes as well are intended for the templates and while using them from javascript is probably possible, you should have a good reason for doing so. Duplication is far cheaper than the wrong abstraction.

Reply:
Well, I think Angular is still quite bad at handling templates, and the templates are by design constructs (no static checks, no compilation time errors). So I'm trying to put as much logic as possible in js. But I accept your answer. Mine was more of a suggestion.